### PR TITLE
ci: refactor tests and benchmarks workflows

### PR DESCRIPTION
**Pull request by antonbaliasnikov**
# What ❔

Refactors tests and benchmarks CI jobs for [CPR-1631](https://linear.app/matterlabs/issue/CPR-1631/fix-branch-name-in-benchmarks-and-integration-testing-workflows-of-era)

## Why ❔

To unblock manual execution and remove code duplications.

## Checklist

- [x] PR title corresponds to the body of PR (we generate changelog entries from PRs).
- [x] Tests for the changes have been added / updated.
- [x] Documentation comments have been added / updated.

### CODE REVIEW

**Comment by antonbaliasnikov** [comment]
@abinavpp , @hedgar2017 , this should be the patch to fix testing and benchmarking jobs and unblock manual testing execution on `0.8.25` branch. I'll refactor `build_and_release` job separately, because it's a bit more complex and requires more attention.

CI is expected to fail, because there is no `solc-bin-zkevm-reference-aba-cpr-1631-fix-branch-name-in-benchmarks-and-integration-testing.json` config file, so it's expected. But after the merge, it should work on `0.8.25`.

File: .github/workflows/tests.yaml
Comment (abinavpp) [inline .github/workflows/tests.yaml]: Does this mean the push event is restricted to the "main" branch? Do we intend to create a "main" branch?

**Review by abinavpp — COMMENTED** [review]
Let's set the target to 0.8.24 since it's the current default.

File: .github/workflows/tests.yaml
Comment (hedgar2017) [reply .github/workflows/tests.yaml]: @antonbaliasnikov there will be no main branch, only the base and PR.

File: .github/workflows/benchmarks.yml
Comment (hedgar2017) [inline .github/workflows/benchmarks.yml]: Is it candidate or any?

**Review by hedgar2017 — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yaml
Comment (antonbaliasnikov) [reply .github/workflows/tests.yaml]: Good catch! Removed this.

File: .github/workflows/benchmarks.yml
Comment (antonbaliasnikov) [reply .github/workflows/benchmarks.yml]: It is any, the LLVM build is always the same here. I thought the thing in this repository was to test the performance impact of `solc` executable, everything else should stay unchanged. At least, it is how it was before, no changes here from my side.

@hedgar2017  could you, please, confirm that it's the right behavior?

**Comment by antonbaliasnikov** [comment]
> Let's set the target to 0.8.24 since it's the current default.

@hedgar2017 , could you, please, confirm that?

File: .github/workflows/benchmarks.yml
Comment (hedgar2017) [reply .github/workflows/benchmarks.yml]: @antonbaliasnikov yes, only the `solc` executable should be changed between `reference` and `candidate` for PRs, but for manual workflows you should also be able to specify the compiler-tester branch and other options that may cause friction if hardcoded.
Sometimes it's beneficial to run a manual workflow against a custom compiler-tester like that with the 0.8.25 support.

My comment was regarding the word `candidate` found several times in this job. To me it looks like the job is generic over reference and candidate, but below I can only see `--benchmark='candidate.json'` and no job to get the `reference.json`. Could you clarify how the job is supposed to work?

**Comment by hedgar2017** [comment]
> > Let's set the target to 0.8.24 since it's the current default.
> 
> @hedgar2017 , could you, please, confirm that?

Probably Abinav meant that it can be problematic to run a manual workflow from a non-default branch (`0.8.25`), and we'll only make it default after the release. So merging into `0.8.24` would suffice at first.
Though I'm not an expert here and I don't object to any approach.

**Review by abinavpp — DISMISSED** [review]

**Comment by dev-Crypto360** [comment]
hey

File: .github/workflows/benchmarks.yml
Comment (antonbaliasnikov) [reply .github/workflows/benchmarks.yml]: @hedgar2017 ,
Got you. This is one line I missed to make generic. Fixed now and added a capability to choose the `compiler-tester` branch as we have in other workflows.

Could you, please, check again? Should be good now.

**Comment by antonbaliasnikov** [comment]
> > > Let's set the target to 0.8.24 since it's the current default.
> > 
> > 
> > @hedgar2017 , could you, please, confirm that?
> 
> Probably Abinav meant that it can be problematic to run a manual workflow from a non-default branch (`0.8.25`), and we'll only make it default after the release. So merging into `0.8.24` would suffice at first. Though I'm not an expert here and I don't object to any approach.

@hedgar2017 , @abinavpp , are you OK if we merge to `0.8.25` and I cherry-pick to `0.8.24` later? When I'm changing the base branch, there are lots of commits appearing.

**Comment by dev-Crypto360** [comment]
👀

**Comment by abinavpp** [comment]
> > > > Let's set the target to 0.8.24 since it's the current default.
> > > 
> > > 
> > > @hedgar2017 , could you, please, confirm that?
> > 
> > Probably Abinav meant that it can be problematic to run a manual workflow from a non-default branch (`0.8.25`), and we'll only make it default after the release. So merging into `0.8.24` would suffice at first. Though I'm not an expert here and I don't object to any approach.
> 
> @hedgar2017 , @abinavpp , are you OK if we merge to `0.8.25` and I cherry-pick to `0.8.24` later? When I'm changing the base branch, there are lots of commits appearing.

Sure. Also, can you squash the commits? Thanks!

**Review by hedgar2017 — APPROVED** [review]

**Comment by abinavpp** [comment]
Cherry-picked to 0.8.24

**Pull request merged**
